docs/api_specs/includes/watcher_api_specs.md

```diff
@@ -0,0 +1,691 @@
+
+```json
+{
+    "event": "unchallenged_exit",
+    "details": {
+        "eth_height"  : 3521678,
+        "utxo_pos"  : 10001001,
+        "owner"  : "B3256026863EB6AE5B06FA396AB09069784EA8EA",
+        "currency"  : "0000000000000000000000000000000000000000",
+        "amount" : 100
+    }
+}
+```
+
+Indicates that an invalid exit is dangerously close to finalization and hasn't been challenged. User should exit.
+
+
+#### `invalid_block`
+> An invalid_block event
+
+```json
+{
+    "event": "invalid_block",
+    "details": {
+        "blockhash"  : "DB32876CC6F26E96B9291682F3AF4A04C2AA2269747839F14F1A8C529CF90225",
+        "blocknum"  : 10000,
+    }
+}
+```
+
+A block containing an invalid tx has been processed [are there other reasons a block can be invalid?]. User should exit.
+
+
+#### `block_withholding`
+> A block_withholding event
+
+```json
+{
+    "event": "block_withholding",
+    "details": {
+        "blockhash"  : "DB32876CC6F26E96B9291682F3AF4A04C2AA2269747839F14F1A8C529CF90225",
+        "blocknum"  : 10000,
+    }
+}
+```
+
+The ChildChain is withholding a block whose hash has been published on the root chain. User should exit.
+
+#### `noncanonical_ife`
+> A noncanonical_ife event
+
+```json
+{
+    "event": "noncanonical_ife",
+    "details": {
+        "inflight_blocknum"  : 10000,
+        "inflight_tx_index"  : 12,
+        "inflight_input_index"  : 1,
+        "competing_blocknum"  : 10000,
+        "competing_tx_index"  : 10,
+        "competing_input_index"  : 0,
+    }
+}
+```
+
+An non-canonical in-flight exit has been started. It should be challenged. 
+<aside class="warning"> Not Implemented Yet.</aside> 
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+inflight_blocknum | integer | Block number of the in-flight transaction
+inflight_tx_index | integer | Index of the in-flight transaction
+inflight_input_index | integer | Index of the double-spent input in the in-flight transaction
+competing_blocknum | integer | Block number of the spending transaction
+competing_tx_index | integer | Index of the spending transaction
+competing_input_index | integer | Index of the double-spent input in the competing transaction
+
+#### `invalid_ife_challenge`
+> A invalid_ife_challenge event
+
+```json
+{
+    "event": "invalid_ife_challenge",
+    "details": {
+        "inflight_blocknum"  : 10000,
+        "inflight_tx_index"  : 12,
+    }
+}
+```
+
+A canonical in-flight exit has been challenged. The challenge should be responded to. 
+<aside class="warning"> Not Implemented Yet.</aside> 
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+inflight_blocknum | integer | Block number of the in-flight transaction
+inflight_tx_index | integer | Index of the in-flight transaction
+
+#### `piggyback_available`
+> A piggyback_available event
+
+```json
+{
+    "event": "piggyback_available",
+    "details": {
+        "inflight_blocknum"  : 10000,
```

Comment:
Not sure if not premature but I have a comment here: 

In order to search for piggybacks that target my address (following our today's conversation and assuming "no filtering" path is picked), this address must be delivered here. 

Probably to be able to get the "piggyback information", caller of `status/` would also need to know whether that is an input or output to piggyback on. There is a number of ways one could architect that, the fundamental thing is that the flow can be followed through by the user.

Reply:
Does the caller need to know whether it's an input or output?
In Rootchain.sol implies you only need the inflight tx bytes and the index:
```
function piggybackInFlightExit(
        bytes _inFlightTx,
        uint8 _outputIndex
    )
```
i.e. the "piggyback information" appears to be the same whether it's an input or an output.

Reply:
ah right, it is because the input vs output is encoded in `_outputIndex`, 0-3 means input, 4-7 means output, assuming 4in 4out tx.

It all depends on what information do you want to serve to the user, I'm not saying explicit input/output flag it's required, it's just worth thinking through and syncing with the "piggyback information" endpoint.

There's however the "only mine" step in the MoreVP flow, where you don't piggyback an input, if you're the only input owner, but maybe we can treat it as optimization? again, just prompting to think through.